preprocessing/extract_seqs_from_cif.py

```
@@ -1,200 +1,239 @@
+            structure = parser.get_structure(pdb_id, handle)
+            chains = {f"{pdb_id}_{chain.id}": seq1(''.join(residue.resname for residue in chain)) for chain in structure.get_chains()}
+        except PDBConstructionException as e:
+            print(f"Error processing file {fname}: {e}. Skipping this file.")
+            return {}  # Return an empty dictionary if an error occurs
+    return chains
+
+
+def write_seqs_from_cifdir(dirpath, fname):
+    structure_dir = Path(dirpath)
+    seq_count = len([name for name in os.listdir(dirpath)])
+    seqs_file = open(fname, "w")
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)